### PR TITLE
more elaborated rpm-macros

### PR DESCRIPTION
**Pull request by besser82**
As title says.  Added some improvements to ` build-tools/rpm/macros.yast` and made it a bit more generic for use on other distris.

### CODE REVIEW

**Comment by mvidner** [comment]
@lslezak please have a look too.

File: build-tools/rpm/macros.yast [hunk 1]
```diff
@@ -15,49 +15,85 @@
...
-%yast_build \
-    %{_prefix}/bin/y2tool y2autoconf \
-    %{_prefix}/bin/y2tool y2automake \
-    autoreconf --force --install \
+# prepare sources for build
+%yast_prep \
+    %{_bindir}/y2tool y2autoconf \
+    %{_bindir}/y2tool y2automake \
+    \
+    # '-Wobsolete' exposes deprecated / obsolete Autotools-macros \
```
Comment (mvidner) [inline build-tools/rpm/macros.yast]: Since we're enabling warnings, any reason why singling out `obsolete`

File: build-tools/rpm/macros.yast [hunk 2]
```diff
@@ -15,49 +15,85 @@
...
+    if [ ! -x "./configure" ]; then \
+        %yast_prep \
+    fi \
+    \
+    # has %%yast_configure been run already? \
+    if [ ! -f "./Makefile" ]; then \
+        %yast_configure \
+    fi \
+    \
+    %make_build
```
Comment (mvidner) [inline build-tools/rpm/macros.yast]: `make_build` is in rpm-4.12.0 but openSUSE-13.2 still has 4.11.3. I have to check whether that matters.

Comment (besser82) [inline build-tools/rpm/macros.yast]: Hmm…  That's a bit strange, somehow…  `%make_build` works fine for me even on (RHEL|CentOS) 6 with rpm-4.8.x.

File: build-tools/rpm/macros.yast (hunk 1 above)
Comment (besser82) [inline build-tools/rpm/macros.yast]: I can change it to `-Wall`, if you'd prefer…  My intention behind `-Wobsolete` was:  Getting information about obsoleted / deprecated Autotools-macros, but not having possible clutter about e.g. portability-stuff…

File: build-tools/rpm/macros.yast [hunk 3]
```diff
@@ -15,49 +15,89 @@
...
 %yast_execcompdir %{_prefix}/lib/YaST2
 %yast_agentdir %{yast_execcompdir}/servers_non_y2
 
-# build the yast module using autotools/make
-%yast_build \
-    %{_prefix}/bin/y2tool y2autoconf \
-    %{_prefix}/bin/y2tool y2automake \
-    autoreconf --force --install \
+# prepare sources for build
+%yast_prep \
```
Comment (lslezak) [inline build-tools/rpm/macros.yast]: Um, so you are splitting the `%yast_build` into `%yast_prep`, `%yast_configure` and `%yast_build`?

Not a bad idea in general, but the drawback is that we would have to update **almost all** yast `*.spec` files to use the new macros. And that's roughly 100 packages... :confused:

Therefore I'm not sure if we should go for this change. What's the advantage (besides splitting a complex macro into smaller simpler parts):question:

Comment (mvidner) [inline build-tools/rpm/macros.yast]: @lslezak That was my objection too at first but then I noticed that the new version of `yast_build` calls the preparatory steps if necessary (※)

File: build-tools/rpm/macros.yast [hunk 4]
```diff
@@ -15,49 +15,89 @@
...
+    \
+    # has %%yast_prep been run already? \
+    if [ ! -x "./configure" ]; then \
+        %yast_prep \
+    fi \
+    \
+    # has %%yast_configure been run already? \
+    if [ ! -f "./Makefile" ]; then \
+        %yast_configure \
+    fi \
```
Comment (mvidner) [inline build-tools/rpm/macros.yast]: ※ here

**Comment by mvidner** [comment]
LGTM.

File: build-tools/rpm/macros.yast (hunk 4 above)
Comment (lslezak) [inline build-tools/rpm/macros.yast]: Oh, I see, it's conditionally called from here... I overlooked that.

**Comment by lslezak** [comment]
Then LGTM also from my side. :+1: 

@besser82 thanks a lot for your contribution, a nice improvement :exclamation:

**Comment by besser82** [comment]
@lslezak and @mvidner: you're welcome!  :D  Thanks for merging!  ^_^

File: build-tools/rpm/macros.yast [hunk 5]
```diff
@@ -15,49 +15,89 @@
...
-%yast_docdir %{_prefix}/share/doc/packages/%{name}
-%yast_logdir /var/log/YaST2
-%yast_vardir /var/lib/YaST2
-%yast_fillupdir /var/adm/fillup-templates
+%yast_libdir %{_libdir}/YaST2
+%yast_plugindir %{yast_libdir}/plugin
+%yast_includedir %{_includedir}/YaST2
+%yast_docdir %{_docdir}/%{name}
+%yast_logdir %{_localstatedir}/log/YaST2
+%yast_vardir %{_sharedstatedir}/YaST2
```
Comment (jreidinger) [inline build-tools/rpm/macros.yast]: @besser82 I have to revert this line change and do it with `%yast_vardir %{_localstatedir}/lib/YaST2`

Reason is that sharedstatedir is by default `/usr/com` and previously it is `/var/lib` so it is not compatible change and can cause us a lot of troubles as store here data and then we cannot read it.

If you need to have it in different location feel free to open pull request, but keep please it working.

Thanks
Josef

File: build-tools/rpm/macros.yast [hunk 6]
```diff
@@ -15,49 +15,89 @@
...
 
 %yast_ybindir %{_prefix}/lib/YaST2/bin
 %yast_ystartupdir %{_prefix}/lib/YaST2
-%yast_plugindir %{_libdir}/YaST2/plugin
-%yast_includedir %{_prefix}/include/YaST2
-%yast_docdir %{_prefix}/share/doc/packages/%{name}
-%yast_logdir /var/log/YaST2
-%yast_vardir /var/lib/YaST2
-%yast_fillupdir /var/adm/fillup-templates
+%yast_libdir %{_libdir}/YaST2
```
Comment (jreidinger) [inline build-tools/rpm/macros.yast]: @besser82 I have to revert it as it is already defined on https://github.com/yast/yast-devtools/pull/88/files#diff-0fc23eb95cef73cdbc8baba0bdc8729aR14 . This way it breaks building of yast packages, os we really want to use real libdir to have `/usr/lib64/YaST2` and yast_libdir should always point to `/usr/lib/YaST2`

**Pull request merged**